.github/workflows/ci.yml

```diff
@@ -0,0 +1,104 @@
+name: Continuous integration
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+
+jobs:
+  test:
+    name: Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+          cache: pip
+
+      - name: Install dependencies
+        run: |
+          pip install -r requirements-dev.txt codecov
+          pip install -e .
+
+      - name: Run tests
+        run: pytest -ra -vv --doctest-modules --cov=.
+
+      - name: Run custom data source tests
+        run: python manage.py test --tag=custom_ds --settings=tests.test_app.settings_custom_ds
+
+      - name: Run custom pk tests
+        run: python manage.py test --tag=custom_pk_ds --settings=tests.test_app.settings_custom_pk_ds
+
+      - name: Coverage
+        run: codecov
+
+  build:
+    name: Build package
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: 3.9
+
+      - name: Install dependencies
+        run: pip install -U setuptools wheel build
+      - name: Build
+        run: python -m build .
+
+  # TODO to be enabled later
+  #black:
+  #  name: Coding style - black
+  #  runs-on: ubuntu-latest
+  #  steps:
+  #    - name: Checkout
+  #      uses: actions/checkout@v2
+  #    - name: Run black
+  #      uses: psf/black@stable
+  #      with:
+  #        args: ". --check"
+
+  #flake8:
+  #  name: Coding style - flake8
+  #  runs-on: ubuntu-latest
+  #  steps:
+  #    - name: Checkout
+  #      uses: actions/checkout@v2
+  #    - name: Set up Python
+  #      uses: actions/setup-python@v2
+  #      with:
+  #        python-version: 3.8
```

Comment:
Mismatching versions (3.9 above, 3.8 here). I know it's commented out, but if I went with the TODO comment and just enabled this later, it might go unnoticed.

Reply:
Well, in hindsight, this is probably not relevant, seeing that I'm suggesting to change the other lines to 3.8 instead...